server.go

```diff
@@ -136,6 +136,13 @@ func (s *Server) ID() uint64 {
 	return s.id
 }
 
+// Index returns the index for the server.
+func (s *Server) Index() uint64 {
+	s.mu.Lock()
```

Comment:
Before we start locking reads of simple 64-bit integers in our code (and if we need to do so, it could be a read lock) we should establish if we *actually need to do so*. Let's be sure we need to lock before locking.